core/payments/payments.go

```diff
@@ -192,6 +192,42 @@ func (s *service) sendEvent(status *pb.RpcMembershipGetStatusResponse) {
 	})
 }
 
+func getCacheExpireTime(dateEnds time.Time) time.Time {
```

Comment:
Maybe it is better to move this logic to separate func of cacheservice?
CacheSet could accept dateEnds except of expireTime, for example, and all internal logics of expireTime calculation will be handled by the very cache